torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -143,7 +143,10 @@ class TORCH_API RpcAgent {
   virtual std::unordered_map<std::string, std::string> getMetrics() = 0;
 
   // Retrive debug info in addition to metrics as KV map
-  virtual std::unordered_map<std::string, std::string> getDebugInfo() = 0;
+  virtual std::unordered_map<std::string, std::string> getDebugInfo();
+
+  // Add GIL wait time data point to metrics
+  virtual void addGilWaitTime(const std::chrono::microseconds gilWaitTime);
```

Comment:
Can we make this private and make PythonRPCHandler a friend class?

Reply:
This means that `GilWaitTimeGuard` where this method is called will have to be the friend class, or we can make `PythonRPCHandler` the friend class and make GilWaitTimeGuard a subclass of `PythonRpcHandler`. Is there a preferred approach?

Reply:
I went with making `GilWaitTimeGuard` a class within `PythonRpcHandler`. If we realize that we need to use it outside of `PythonRpcHandler`, we can consider a different approach.